### DISCUSSION TITLE
Running Test Files

### DISCUSSION THREAD

**Discussion by kennedy12335** [pykale/pykale#180] [Q&A] [answered]
I am currently trying to run the test_csv_logger.py file and I am encountering an error. It cannot locate files from the folder containing the test data. I believe the test data is downloaded automatically from the confest.py script and I assume that is when the test_data folder would be created. I may be wrong but that was what I understood from https://github.com/pykale/pykale/blob/main/tests/README.md. Can someone please explain why I am getting errors. To be precise this is the error I am getting:

_E           FileNotFoundError: [Errno 2] No such file or directory: 'tests/test_data/parameters.json'

kale/utils/csv_logger.py:66: FileNotFoundError_

**Comment by bobturneruk** [comment, marked as answer]
Hi @kennedy12335 - can you share what line of code you used to run the test, please?

**Reply by kennedy12335** [reply]
The code I used to run the test is :

pytest tests/utils/test_csv_logger.py -v

I set main to be the root folder hence the use of the path when specifying the test file.

**Reply by bobturneruk** [reply]
What OS, please?

**Reply by haipinglu** [reply]
@kennedy12335 Please use the existing template https://github.com/pykale/pykale/issues/new?assignees=&labels=bug&template=bug-report.md&title=%5BBug%5D to report the encountered problem, for efficiency and reproducibility.

**Reply by kennedy12335** [reply]
I'm on a mac and the python interpreter is conda.

**Reply by bobturneruk** [reply]
Cheers. If I delete my `test_data` folder I can replicate.

**Reply by bobturneruk** [reply]
I'll follow up in the bug issue as per @haipinglu comment.

**Reply by kennedy12335** [reply]
I have reported the bug issue. I was not sure the problem could be classified as a bug hence why I asked first in the discussion. The link to the issue is https://github.com/pykale/pykale/issues/183
